### PR TITLE
feature/CPLP-1831-getapplication-refactoring-companyrole

### PR DESCRIPTION
**Pull request by VPrasannaK94**
https://jira.catena-x.net/browse/CPLP-1831

### CODE REVIEW

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyApplicationDetails.cs [hunk 1]
```diff
@@ -48,6 +48,8 @@ public CompanyApplicationDetails(Guid applicationId, CompanyApplicationStatusId
     public string? BusinessPartnerNumber { get; set; }
     [JsonPropertyName("documents")]
     public IEnumerable<DocumentDetails> Documents { get; set; }
+    [JsonPropertyName("companyRoles")]
+    public IEnumerable<string>? CompanyRoles { get; set; }
```
Comment (qxz2mqe) [inline src/portalbackend/PortalBackend.DBAccess/Models/CompanyApplicationDetails.cs:52]: Remove nullable. change it from nullable to non nullable

File: src/administration/Administration.Service/Controllers/RegistrationController.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ public RegistrationController(IRegistrationBusinessLogic logic)
     /// <param name="companyName">search by company name</param>
     /// <returns>Company Application Details</returns>
     /// <remarks>
-    /// Example: GET: api/administration/registration/applications?companyName=Car&amp;page=0&amp;size=4 <br />
+    /// Example: GET: api/administration/registration/applications?companyName=Car&amp;page=0&amp;size=4&amp;companyApplicationStatus=Closed <br />
```
Comment (qxz2mqe) [inline src/administration/Administration.Service/Controllers/RegistrationController.cs:74]: why this filter is required here?

File: tests/administration/Administration.Service.Tests/BusinessLogic/RegistrationBusinessLogicTest.cs [hunk 3]
```diff
@@ -223,11 +223,12 @@ public async Task GetCompanyApplicationDetailsAsync_WithDefaultRequest_GetsExpec
 
         // Act
         var result = await _logic.GetCompanyApplicationDetailsAsync(0, 5,null).ConfigureAwait(false);
-
```
Comment (qxz2mqe) [inline tests/administration/Administration.Service.Tests/BusinessLogic/RegistrationBusinessLogicTest.cs:226]: Remove all blank spaces

File: tests/administration/Administration.Service.Tests/BusinessLogic/RegistrationBusinessLogicTest.cs [hunk 4]
```diff
@@ -227,7 +227,7 @@ public async Task GetCompanyApplicationDetailsAsync_WithDefaultRequest_GetsExpec
         // Assert
         A.CallTo(() => _applicationRepository.GetCompanyApplicationsFilteredQuery(null, A<IEnumerable<CompanyApplicationStatusId>>.That.Matches(x => x.Count() == 3 && x.All(y => companyAppStatus.Contains(y))))).MustHaveHappenedOnceExactly();
         Assert.IsType<Pagination.Response<CompanyApplicationDetails>>(result);
-        result.Content.Should().HaveCount(5);
+        result.Content.Should().HaveCount(5);
```
Comment (Phil91) [inline tests/administration/Administration.Service.Tests/BusinessLogic/RegistrationBusinessLogicTest.cs:230]: please remove whitespaces

File: src/registration/Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs [hunk 5]
```diff
@@ -421,8 +421,7 @@ public async Task<bool> SubmitRegistrationAsync(Guid applicationId, string iamUs
         if (applicationUserData.CompanyUserId == Guid.Empty)
         {
             throw new ForbiddenException($"iamUserId {iamUserId} is not assigned with CompanyApplication {applicationId}");
-        }
-
+        }
```
Comment (Phil91) [inline src/registration/Registration.Service/BusinessLogic/RegistrationBusinessLogic.cs:424]: please add a linebreak

File: src/portalbackend/PortalBackend.DBAccess/Repositories/DocumentRepository.cs [hunk 6]
```diff
@@ -118,5 +118,5 @@ public Document CreateDocument(string documentName, byte[] documentContent, byte
                 x.Applications.Any(companyApplication => companyApplication.Company!.CompanyUsers.Any(companyUser => companyUser.IamUser!.UserEntityId == iamUserId)),
                 x.Document.DocumentTypeId,
                 x.Applications.Any(companyApplication => applicationStatusIds.Contains(companyApplication.ApplicationStatusId))))
-            .SingleOrDefaultAsync();
+            .SingleOrDefaultAsync();
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/DocumentRepository.cs:121]: please remove whitespaces

**Review by Phil91 — CHANGES_REQUESTED** [review]
Only small formatting issues

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyApplicationDetails.cs [hunk 7]
```diff
@@ -48,6 +48,8 @@ public CompanyApplicationDetails(Guid applicationId, CompanyApplicationStatusId
     public string? BusinessPartnerNumber { get; set; }
     [JsonPropertyName("documents")]
     public IEnumerable<DocumentDetails> Documents { get; set; }
+    [JsonPropertyName("companyRoles")]
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Models/CompanyApplicationDetails.cs:51]: Please add the CompanyRoles to the constructor. If they are optional initialize with =new List<string>()

**Review by Phil91 — CHANGES_REQUESTED** [review]

**Review by Phil91 — APPROVED** [review]

**Pull request merged**